test/system/deposit-validation.test.js

```diff
@@ -0,0 +1,134 @@
+  const collateralizationThreshold = 300
+
+  let tbtcToken
+  let underwriterToken
+  let assetPool
+  let signerBondsSwapStrategy
+  let coveragePool
+  let riskManagerV1
+  let fakeDeposit
+
+  let governance
+  let rewardsManager
+  let thirdParty
+
+  before(async () => {
+    await resetFork(startingBlock)
+
+    governance = await ethers.getSigner(0)
+    rewardsManager = await ethers.getSigner(1)
+    thirdParty = await impersonateAccount(thirdPartyAddress)
+
+    tbtcToken = await ethers.getContractAt("IERC20", tbtcTokenAddress)
+
+    const UnderwriterToken = await ethers.getContractFactory("UnderwriterToken")
+    underwriterToken = await UnderwriterToken.deploy("Coverage KEEP", "covKEEP")
+    await underwriterToken.deployed()
+
+    const AssetPool = await ethers.getContractFactory("AssetPool")
+    assetPool = await AssetPool.deploy(
+      keepTokenAddress,
+      underwriterToken.address,
+      rewardsManager.address
+    )
+    await assetPool.deployed()
+    await underwriterToken.transferOwnership(assetPool.address)
+
+    const SignerBondsSwapStrategy = await ethers.getContractFactory(
+      "SignerBondsEscrow"
+    )
+    signerBondsSwapStrategy = await SignerBondsSwapStrategy.deploy()
+    await signerBondsSwapStrategy.deployed()
+
+    const Auction = await ethers.getContractFactory("Auction")
+
+    const masterAuction = await Auction.deploy()
+    await masterAuction.deployed()
+
+    const CoveragePool = await ethers.getContractFactory("CoveragePool")
+    coveragePool = await CoveragePool.deploy(assetPool.address)
+    await coveragePool.deployed()
+    await assetPool.transferOwnership(coveragePool.address)
+
+    const RiskManagerV1 = await ethers.getContractFactory("RiskManagerV1")
+    riskManagerV1 = await RiskManagerV1.deploy(
+      tbtcToken.address,
+      tbtcDepositTokenAddress,
+      coveragePool.address,
+      signerBondsSwapStrategy.address,
+      masterAuction.address,
+      auctionLength,
+      collateralizationThreshold
+    )
+    await riskManagerV1.deployed()
+
+    await coveragePool
+      .connect(governance)
+      .beginRiskManagerApproval(riskManagerV1.address)
+    await increaseTime(2592000) // +30 days
+    await coveragePool
+      .connect(governance)
+      .finalizeRiskManagerApproval(riskManagerV1.address)
+
+    // Suppose a third party deploys an arbitrary deposit contract.
+    // For simplicity, let's say it's just the DepositStub.
+    const DepositStub = await ethers.getContractFactory("DepositStub")
+    fakeDeposit = await DepositStub.connect(thirdParty).deploy(
+      tbtcToken.address,
+      to1e18(1)
+    )
+    await fakeDeposit.deployed()
+  })
+
+  describe("initial state", () => {
+    it("should assert the fake deposit's auction does not exist", async () => {
+      const auctionAddress = await riskManagerV1.depositToAuction(
+        fakeDeposit.address
+      )
+      expect(auctionAddress).to.be.equal(ZERO_ADDRESS)
+    })
+  })
```

Comment:
It feels weird initial state assertion is inside a  describe... 🤔 What if we had `assertInitialState` function called at the end of `before` setup, or having initial state assertions inside `beforeEach`?

Reply:
This is what we do in `system/liquidation.test.js` as well. In my opinion `describe initial state` sounds good and shows explicitly what state should we expect at the beginning. I think moving that to `assertInitialState` will hide those details and make the scenario harder to understand.

Reply:
What would you say about:
```
  describe("test initial state", () => {
    describe("fake deposit's auction", () => {
      it("should not exist", async () => {
        const auctionAddress = await riskManagerV1.depositToAuction(
          fakeDeposit.address
        )
        expect(auctionAddress).to.be.equal(ZERO_ADDRESS)
      })
    })
  })
```

and for `system/liquidation.test.js`:
```
  describe("test initial state", () => {
    describe("deposit", () => {
      it("should be in active state", async () => {
        expect(await tbtcDeposit.currentState()).to.equal(5) // Active
      })
    })

    describe("auction", () => {
      it("should not exist", async () => {
        const auctionAddress = await riskManagerV1.depositToAuction(
          tbtcDeposit.address
        )
        expect(auctionAddress).to.be.equal(ZERO_ADDRESS)
      })
    })
  })
```

There are two things that feels wrong to me with the current structure:
- We are describing tests's expected initial state after the fork, not the deposit or auction initial state after it's been created. For a moment I was confused about it when reading the code for the first time.
- `it("should assert...` does not match the approach we use in other tests.

Reply:
Sounds good. Please see https://github.com/keep-network/coverage-pools/pull/77/commits/1e6ae20db02b7b0fac882d941e9704f9d56eda46